torchok/models/backbones/swin.py

```diff
@@ -1,182 +1,143 @@
 """TorchOK Swin Transformer V2
+A PyTorch implementation of `Swin Transformer V2: Scaling Up Capacity and Resolution`
+    - https://arxiv.org/abs/2111.09883
 
 Adapted from https://github.com/microsoft/Swin-Transformer/blob/main/models/swin_transformer_v2.py
-Copyright (c) 2022 Microsoft
-Licensed under The MIT License [see LICENSE for details]
+(Copyright (c) 2022 Microsoft)
+and from https://github.com/rwightman/pytorch-image-models/blob/master/timm/models/swin_transformer_v2.py
+(Copyright 2022, Ross Wightman)
+Licensed under Apache License 2.0 [see LICENSE for details]
 """
-# --------------------------------------------------------
-# Swin Transformer V2
-# Copyright (c) 2022 Microsoft
-# Licensed under The MIT License [see LICENSE for details]
-# Written by Ze Liu
-# --------------------------------------------------------
+
+from typing import List
+
 import torch
 import torch.nn as nn
 import torch.utils.checkpoint as checkpoint
-from typing import List, Tuple, Union, Optional
+from timm.data import IMAGENET_DEFAULT_MEAN, IMAGENET_DEFAULT_STD
+from timm.models.helpers import build_model_with_cfg
+from timm.models.layers import trunc_normal_
+from timm.models.swin_transformer_v2 import BasicLayer as SwinBasicLayer, checkpoint_filter_fn, PatchEmbed, PatchMerging
 
 from torchok.constructor import BACKBONES
 from torchok.models.backbones.base_backbone import BaseBackbone
-from torchok.models.modules.weights_init import trunc_normal_
-from torchok.models.modules.blocks.patch_merging import PatchMerging
-from torchok.models.modules.blocks.patch_embedding import PatchEmbed
-from torchok.models.modules.blocks.swin_block import SwinTransformerBlock
-from torchok.models.backbones.utils.helpers import build_model_with_cfg
 
 
-# The weights was taken from https://github.com/rwightman/pytorch-image-models
-# Licensed under The Apache 2.0 License [see LICENSE for details]
+def _cfg(url='', **kwargs):
+    return {
+        'url': url, 'input_size': (3, 224, 224), 'pool_size': None,
+        'crop_pct': .9, 'interpolation': 'bicubic', 'fixed_input_size': True,
+        'mean': IMAGENET_DEFAULT_MEAN, 'std': IMAGENET_DEFAULT_STD,
+        'first_conv': 'patch_embed.proj', 'classifier': 'head',
+        **kwargs
+    }
+
+
 default_cfgs = {
-    'swinv2_tiny_window8_256': dict(
-        url='https://torchok-hub.s3.eu-west-1.amazonaws.com/swinv2_tiny_patch4_window8_256.pth'
+    'swinv2_tiny_window8_256': _cfg(
+        url='https://github.com/SwinTransformer/storage/releases/download/v2.0.0/swinv2_tiny_patch4_window8_256.pth',
```

Comment:
Why not use base_checkpoint_url like in HRNet?

Reply:
Added